### PR TITLE
pcrlock: add simple Varlink API + some varlinkctl tweaks

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: src/pcrlock/pcrlock.c [hunk 1]
```diff
@@ -2429,66 +2498,15 @@ static int verb_show_cel(int argc, char *argv[], void *userdata) {
...
-                                         JSON_BUILD_OBJECT(
-                                                         JSON_BUILD_PAIR_UNSIGNED("pcr", record->pcr),
-                                                         JSON_BUILD_PAIR_UNSIGNED("recnum", ++recnum),
-                                                         JSON_BUILD_PAIR_VARIANT("digests", ja),
-                                                         JSON_BUILD_PAIR_CONDITION(ct, "content_type", JSON_BUILD_STRING(ct)),
-                                                         JSON_BUILD_PAIR_CONDITION(cd, "content", JSON_BUILD_VARIANT(cd))));
+                r = json_variant_append_array(&array, cel);
                 if (r < 0)
-                        return log_error_errno(r, "Failed to append CEL record: %m");
+                        return log_error_errno(r, "Failed append CEL record: %m");
```
Comment (bluca) [inline src/pcrlock/pcrlock.c]: "Failed to append" seemed more correct to me?

File: src/pcrlock/pcrlock.c [hunk 2]
```diff
@@ -5133,6 +5234,34 @@ static int run(int argc, char *argv[]) {
         if (r <= 0)
                 return r;
 
+        if (arg_varlink) {
+                _cleanup_(varlink_server_unrefp) VarlinkServer *varlink_server = NULL;
+
+                /* Invocation as Varlink service */
+
+                r = varlink_server_new(&varlink_server, VARLINK_SERVER_ROOT_ONLY);
```
Comment (bluca) [inline src/pcrlock/pcrlock.c:5242]: What about polkit integration? At least for reading logs it seems useful to have?

**Review by bluca — APPROVED** [review]

File: src/pcrlock/pcrlock.c (hunk 2 above)
Comment (poettering) [reply src/pcrlock/pcrlock.c:5242]: Yeah, but let's add that later. We can open this up if we want to.

It's not entirely clear why we would not just make it world readable though, which is why i didn't add this yet.

Comment (bluca) [reply src/pcrlock/pcrlock.c:5242]: Ok, can you list it in the TODO so we don't forget?

**Pull request merged**